src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -4285,131 +4286,135 @@ void SetupAdaptiveConvectionStaticMetaData(EnergyPlusData &state)
-                state.dataConvectionCoefficient->RoofGeo.XuYdZu.Vertex = Surface(SurfLoop).Vertex(1);
-
-                state.dataConvectionCoefficient->RoofGeo.XuYuZu.SurfNum = SurfLoop;
-                state.dataConvectionCoefficient->RoofGeo.XuYuZu.VertNum = 1;
-                state.dataConvectionCoefficient->RoofGeo.XuYuZu.Vertex = Surface(SurfLoop).Vertex(1);
+            if (state.dataConvectionCoefficient->FirstRoofSurf) {
+                // #9432 - We can't just Init with anything in the group in the group (such as Vertex(1))
+                // Otherwise if for eg the Vertex(1) has a higher X and Y than all other points, then we will not be able
+                // to find the XdYu ones (lo X, high Y), that is the top left corner because
+                //     vertex.x <= RoofGeo.XdYuZd.Vertex.x => True
+                //     vertex.y => RoofGeo.XdYuZd.Vertex.y => False
+                // So instead, we initialize to the Surface's Centroid...
+                RoofGeo.XdYdZd.SurfNum = SurfLoop;
+                RoofGeo.XdYdZd.VertNum = 0;
+                RoofGeo.XdYdZd.Vertex = Surface(SurfLoop).Centroid;
+
+                RoofGeo.XdYdZu.SurfNum = SurfLoop;
+                RoofGeo.XdYdZu.VertNum = 0;
+                RoofGeo.XdYdZu.Vertex = Surface(SurfLoop).Centroid;
+
+                RoofGeo.XdYuZd.SurfNum = SurfLoop;
+                RoofGeo.XdYuZd.VertNum = 0;
+                RoofGeo.XdYuZd.Vertex = Surface(SurfLoop).Centroid;
+
+                RoofGeo.XdYuZu.SurfNum = SurfLoop;
+                RoofGeo.XdYuZu.VertNum = 0;
+                RoofGeo.XdYuZu.Vertex = Surface(SurfLoop).Centroid;
+
+                RoofGeo.XuYdZd.SurfNum = SurfLoop;
+                RoofGeo.XuYdZd.VertNum = 0;
+                RoofGeo.XuYdZd.Vertex = Surface(SurfLoop).Centroid;
+
+                RoofGeo.XuYuZd.SurfNum = SurfLoop;
+                RoofGeo.XuYuZd.VertNum = 0;
+                RoofGeo.XuYuZd.Vertex = Surface(SurfLoop).Centroid;
+
+                RoofGeo.XuYdZu.SurfNum = SurfLoop;
+                RoofGeo.XuYdZu.VertNum = 0;
+                RoofGeo.XuYdZu.Vertex = Surface(SurfLoop).Centroid;
+
+                RoofGeo.XuYuZu.SurfNum = SurfLoop;
+                RoofGeo.XuYuZu.VertNum = 0;
+                RoofGeo.XuYuZu.Vertex = Surface(SurfLoop).Centroid;
 
                 state.dataConvectionCoefficient->FirstRoofSurf = false;
             }
             // treat as part of roof group
-            state.dataConvectionCoefficient->RoofGeo.Area += thisArea;
+            RoofGeo.Area += thisArea;
+
+            // I take the double by reference to avoid adding a rounding error on top of a rounding error (the centroid is calculated...)
+            auto lesserThanWithTol = [](const auto &lhs, const auto &rhs) { return lhs - rhs <= 0.000001; };
+            auto greaterThanWithTol = [](const auto &lhs, const auto &rhs) { return lhs - rhs >= -0.000001; };
```

Comment:
Gotta bake in some tolerance... The centroid is calculated. `TestInitHBDaylightingNoExtWindow` was failing because of that.

Actual vertex 'z' in the test: 12.9774
In the debugger, Surface(SurfLoop).Vertex(VertLoop).z: 12.977399999999999
Centroid.z = 12.977399999999996